autoload/neomake/makers/ft/python.vim

```diff
@@ -296,3 +296,45 @@ function! neomake#makers#ft#python#py3kwarn() abort
         \ 'errorformat': '%W%f:%l:%c: %m',
         \ }
 endfunction
+
+" Find out the python version to be used in the current buffer (script)
+" by checking the shebang. If no shebangis found or it does not conatain
+" an explicit version number, use the default interpreter.
+" If the shebang changed since the last write, unset the buffer specific
+" settings, then set the new values.
+function! neomake#makers#ft#python#SelectAlternatives() abort
+
+    if !exists('s:default_version')
+        " Get defult python executable version as [major, minor, revision]
+        let s:default_version = split(matchstr(system('python --version'), '[0-9\.]\+'), '\')
+    endif
+
+    " Scan the shebang
+    let l:shebang = matchlist(getline(1), '^#!.*python\([0-9.]*\)')
+
+    if empty(l:shebang) || l:shebang[1] == ''
+        " Use the major number from s:default_version
+        " Ip $ the shebang does not contain one or is absent
+        let l:pyversion = s:default_version[0]
+    else
+        let l:pyversion = l:shebang[1]
```

Comment:
This could also call it with `--version`, instead of relying on the name.
Maybe only in case of `l:shebang[1] == ''`?